front/pages/api/w/[wId]/assistant/agent_configurations/[aId]/index.ts

```diff
@@ -110,44 +109,37 @@ async function handler(
         });
       }
 
-      let agentConfigurationRes: Result<AgentConfigurationType, Error>;
       const maxToolsUsePerRun =
         bodyValidation.right.assistant.maxToolsUsePerRun;
 
-      if (!bodyValidation.right.useMultiActions) {
-        if (maxToolsUsePerRun !== undefined) {
-          return apiError(req, res, {
-            status_code: 400,
-            api_error: {
-              type: "app_auth_error",
-              message:
-                "maxToolsUsePerRun is only supported in multi-actions mode.",
-            },
-          });
-        }
-        agentConfigurationRes = await createOrUpgradeAgentConfiguration({
-          auth,
-          assistant: { ...bodyValidation.right.assistant, maxToolsUsePerRun },
-          legacySingleActionMode: true,
-          agentConfigurationId: req.query.aId as string,
+      const isLegacyConfiguration =
+        bodyValidation.right.assistant.actions.length === 1 &&
+        !bodyValidation.right.assistant.actions[0].description;
+
+      if (isLegacyConfiguration && maxToolsUsePerRun !== undefined) {
```

Comment:
Sanity check: Are we sure that the assistant builder for legacy agent configuration does not save the `maxToolsUsePerRun` cause looking below it's not explicitly set to `undefined`?

Reply:
I think it's okay since we're not using this field when legacyAgent on the multiActionRunLoop